File: test/unit/data/http_log_level_test.dart
import 'package:exponea/exponea.dart';
import 'package:exponea/src/data/encoder/main.dart';
import 'package:flutter_test/flutter_test.dart';

import 'base.dart';

void main() {
  group('HttpLoggingLevel', () {
    const encode = HttpLoggingLevelEncoder.encode;
    const decode = HttpLoggingLevelEncoder.decode;

    final data = readStringData('http_log_level');
    test('check data', () async {
      expect(data.length, 4);
    });

    test('encode', () async {
      for (int i = 0; i < data.length; i++) {
        expect(encode(HttpLoggingLevel.values[i]), data[i]);
      }
    });

    test('decode', () async {
      for (int i = 0; i < data.length; i++) {
        expect(decode(data[i]), HttpLoggingLevel.values[i]);
      }
      expect(() => decode('FOO'), throwsUnsupportedError);
    });
  });
}

File: example/android/app/src/test/kotlin/com/exponea/example/HttpLogLevelTest.kt
package com.exponea.example

import com.exponea.sdk.models.ExponeaConfiguration.HttpLoggingLevel
import org.junit.Assert.assertEquals
import org.junit.Assert.fail
import org.junit.BeforeClass
import org.junit.Test

class HttpLogLevelTest {
    companion object {
        lateinit var data: List<String>

        @BeforeClass @JvmStatic fun setup() {
            data = BaseTest.readStringData("http_log_level")
        }
    }

    @Test
    fun `validate data`() {
        assertEquals(data.size, 4)
    }

    @Test
    fun `encode`() {
        for (i in 0..(data.size - 1)) {
            assertEquals(HttpLoggingLevel.values()[i].name, data[i])
        }
    }

    @Test
    fun `decode`() {
        for (i in 0..(data.size - 1)) {
            assertEquals(HttpLoggingLevel.valueOf(data[i]), HttpLoggingLevel.values()[i])
        }
    }
}
